Source/EasyNetQ/Producer/Rpc.cs

```diff
@@ -229,11 +229,18 @@ protected virtual void RequestPublish<TRequest>(TRequest request, string routing
             var routingKey = conventions.RpcRoutingKeyNamingConvention(typeof(TRequest));
 
             var exchange = advancedBus.ExchangeDeclare(conventions.RpcExchangeNamingConvention(), ExchangeType.Direct);
-            var queue = advancedBus.QueueDeclare(routingKey);
+            var queue = advancedBus.QueueDeclare(routingKey, durable: configuration.Durable);
             advancedBus.Bind(exchange, queue, routingKey);
 
             return advancedBus.Consume<TRequest>(queue, (requestMessage, messageRecievedInfo) => ExecuteResponder(responder, requestMessage),
-                c => c.WithPrefetchCount(configuration.PrefetchCount));
+                c => c.WithPrefetchCount(configuration.PrefetchCount).WithRecoveryAction(() =>
+                    {
+                        //just in case we loose exchange and/or queue in case of owning cluster node went down, and they are nondurable
```

Comment:
in fact this code is causing an exception when connection fails over the slave node:

```
RabbitMQ.Client.Exceptions.OperationInterruptedException: The AMQP operation was interrupted: AMQP close-reason, initiated by Peer, code=404, text="NOT_FOUND - home node 'rabbit-01@HOME-WS' of durable queue 'EasyNetQ.TestingRecovery.MyRequest:EasyNetQ.TestingRecovery' in vhost '/' is down or inaccessible", classId=50, methodId=10, cause=
   at EasyNetQ.Producer.ClientCommandDispatcherSingleton.Invoke[T](Func`2 channelAction) in C:\Users\micde\Documents\Git\EasyNetQ\Source\EasyNetQ\Producer\ClientCommandDispatcherSingleton.cs:line 39
   at EasyNetQ.Producer.ClientCommandDispatcher.Invoke[T](Func`2 channelAction) in C:\Users\micde\Documents\Git\EasyNetQ\Source\EasyNetQ\Producer\ClientCommandDispatcher.cs:line 28
   at EasyNetQ.RabbitAdvancedBus.QueueDeclare(String name, Boolean passive, Boolean durable, Boolean exclusive, Boolean autoDelete, Nullable`1 perQueueMessageTtl, Nullable`1 expires, Nullable`1 maxPriority, String deadLetterExchange, String deadLetterRoutingKey, Nullable`1 maxLength, Nullable`1 maxLengthBytes) in C:\Users\micde\Documents\Git\EasyNetQ\Source\EasyNetQ\RabbitAdvancedBus.cs:line 400
   at EasyNetQ.Producer.Rpc.<>c__DisplayClass22_0`2.<Respond>b__2() in C:\Users\micde\Documents\Git\EasyNetQ\Source\EasyNetQ\Producer\Rpc.cs:line 241
   at EasyNetQ.Consumer.PersistentConsumer.StartConsumingInternal() in C:\Users\micde\Documents\Git\EasyNetQ\Source\EasyNetQ\Consumer\PersistentConsumer.cs:line 76
```